src/dbus_api.rs

```diff
@@ -286,7 +284,60 @@ fn destroy_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 }
 
 fn list_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
-    Ok(vec![m.msg.method_return().append3("/dbus/cache/path", 0, "Ok")])
+    let message: &Message = m.msg;
+    let mut iter = message.iter_init();
+
+    if iter.arg_type() != 0 {
```

Comment:
Ok, the only other completed method that takes no arguments is list_pools.  I updated it.

I think the check makes sense. But if folks disagree, I'll take them out.